File: JavaScript/Ordenacao.js
console.log("Ordenação de números (Quicksort).");
let n = parseInt(prompt("Digite quantos números deseja ordenar: "));

let numeros = [];
for (let i = 0; i < n; i++) {
    numeros.push(parseInt(prompt("Digite o número " + (i + 1) + ": ")));
}

function particionar(arr, inicio, fim) {
    let pivo = arr[fim];
    let i = inicio - 1;

    for (let j = inicio; j < fim; j++) {
        if (arr[j] <= pivo) {
            i++;
            let temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }
    }

    let temp = arr[i + 1];
    arr[i + 1] = arr[fim];
    arr[fim] = temp;

    return i + 1;
}

function quicksort(arr, inicio, fim) {
    if (inicio < fim) {
        let posPivo = particionar(arr, inicio, fim);
        quicksort(arr, inicio, posPivo - 1);
        quicksort(arr, posPivo + 1, fim);
    }
}

quicksort(numeros, 0, numeros.length - 1);

console.log("Números ordenados: " + numeros.join(" ") + "\n");

File: java/Ordenacao.java
import java.util.Scanner;

public class Ordenacao {
    void ordenarNumeros(){
        Scanner dado = new Scanner(System.in);

        System.out.println("Ordenação de números (Quicksort).");
        System.out.println("Digite quantos números deseja ordenar: ");
        int n = dado.nextInt();

        int[] numeros = new int[n];

        for (int i = 0; i < n; i++) {
            System.out.println("Digite o número " + (i + 1) + ": ");
            numeros[i] = dado.nextInt();
        }

        quicksort(numeros, 0, n - 1);

        System.out.print("Números ordenados: ");
        for(int num : numeros){
            System.out.print(num + " ");
        }
        System.out.println("\n");
    }

    private void quicksort(int[] arr, int inicio, int fim){
        if (inicio < fim) {
            int posPivo = particionar(arr, inicio, fim);
            quicksort(arr, inicio, posPivo - 1);
            quicksort(arr, posPivo + 1, fim);
        }
    }

    private int particionar(int[] arr, int inicio, int fim){
        int pivo = arr[fim];
        int i = inicio - 1;

        for (int j = inicio; j < fim; j++) {
            if (arr[j] <= pivo) {
                i++;
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }

        int temp = arr[i + 1];
        arr[i + 1] = arr[fim];
        arr[fim] = temp;

        return i + 1;
    }
}